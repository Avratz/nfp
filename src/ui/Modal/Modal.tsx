'use client'
import { useRef, useEffect, ReactNode } from 'react'
import { createPortal } from 'react-dom'

import styles from './Modal.module.css'

interface ModalProps {
  children: ReactNode
  closeModal: () => void
}

export function Modal({ children, closeModal }: ModalProps) {
  const modalElement = useRef<Element | null>(null)

  if (!modalElement.current) {
    modalElement.current = document.createElement('div')
  }

  useEffect(() => {
    const modalRoot = document.getElementById('modal')
    if (modalRoot !== null && modalElement.current !== null) {
      modalRoot.appendChild(modalElement.current)
    }

    return () => {
      if (modalRoot !== null && modalElement.current !== null) {
        modalRoot.removeChild(modalElement.current)
      }
    }
  }, [])

  return createPortal(
    <div className={styles.overlay} onClick={closeModal}>
      {children}
    </div>,
    modalElement.current,
  )
}
