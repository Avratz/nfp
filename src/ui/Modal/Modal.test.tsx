import { render } from '@testing-library/react'
import { Modal } from './Modal'

describe('Modal', () => {
  it('renders default', () => {
    const { container } = render(<Modal closeModal={() => null}>Modal</Modal>)
    expect(container).toMatchSnapshot()
  })
})
