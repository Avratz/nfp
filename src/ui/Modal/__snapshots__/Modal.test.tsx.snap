// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders default 1`] = `<div />`;
